### PR TITLE
target/riscv: support log memory access128 for read

### PR DESCRIPTION
**Pull request by zqb-all**
Change-Id: I9235150fa00c03a1d75d0b44a7500758daa56e2b

### CODE REVIEW

File: src/target/riscv/riscv-013.c [hunk 1]
```diff
@@ -2831,33 +2839,50 @@ static int read_memory_bus_v1(struct target *target, target_addr_t address,
...
+					if (next_read_j == 0) {
+						if (size == 16) {
+							uint64_t value_h = ((uint64_t)sbvalue[3] << 32) | sbvalue[2];
+							uint64_t value_l = ((uint64_t)sbvalue[1] << 32) | sbvalue[0];
+							log_memory_access128(next_read, value_h, value_l, true);
+						} else {
+							uint64_t value = ((uint64_t)sbvalue[1] << 32) | sbvalue[0];
+							log_memory_access(next_read, value, size, true);
+						}
+						memset(sbvalue, 0, size);
```
Comment (zqb-all) [inline src/target/riscv/riscv-013.c:2820]: I'm not sure whether to keep this memset, can anyone give me some advice

File: src/target/riscv/riscv-013.c [hunk 2]
```diff
@@ -2831,33 +2839,50 @@ static int read_memory_bus_v1(struct target *target, target_addr_t address,
...
+					buf_set_u32(buffer + next_read - address, 0, 8 * MIN(size, 4), sbvalue[next_read_j]);
+					if (next_read_j == 0) {
+						if (size == 16) {
+							uint64_t value_h = ((uint64_t)sbvalue[3] << 32) | sbvalue[2];
+							uint64_t value_l = ((uint64_t)sbvalue[1] << 32) | sbvalue[0];
+							log_memory_access128(next_read, value_h, value_l, true);
+						} else {
+							uint64_t value = ((uint64_t)sbvalue[1] << 32) | sbvalue[0];
+							log_memory_access(next_read, value, size, true);
+						}
```
Comment (timsifive) [inline src/target/riscv/riscv-013.c]: Can you refactor this into a function that takes sbvalue and size, so the code can be shared with the same code below?

File: src/target/riscv/riscv-013.c (hunk 1 above)
Comment (timsifive) [reply src/target/riscv/riscv-013.c:2820]: Keep it. If it turns out to slow things down (I doubt it), then we can put this whole block inside a check so it only runs when debug is enabled.

**Review by timsifive — CHANGES_REQUESTED** [review]

File: src/target/riscv/riscv-013.c (hunk 2 above)
Comment (zqb-all) [reply src/target/riscv/riscv-013.c]: sure, just submitted the new patch

**Review by timsifive — APPROVED** [review]

**Pull request merged**